Plots/Contours/NCL_eof_1_1.py

```diff
@@ -0,0 +1,326 @@
+    endDate = xMon.time[-1]
+    seasons_pd = {'DJF': ('QS-DEC', 1), 'JFM': ('QS-JAN',  2), 'FMA': ('QS-FEB',  3), 'MAM': ('QS-MAR',  4),
+                  'AMJ': ('QS-APR', 5), 'MJJ': ('QS-MAY',  6), 'JJA': ('QS-JUN',  7), 'JAS': ('QS-JUL',  8),
+                  'ASO': ('QS-AUG', 9), 'SON': ('QS-SEP', 10), 'OND': ('QS-OCT', 11), 'NDJ': ('QS-NOV', 12)}
+    try:
+        (season_pd, season_sel) = seasons_pd[season]
+    except KeyError:
+        raise ValueError("contributed: month_to_season: bad season: SEASON = " + season)
+
+    # Compute the three-month means, moving time labels ahead to the middle month.
+    month_offset = 'MS'
+    xSeasons = xMon.resample(time=season_pd, loffset=month_offset).mean()
+
+    # Filter just the desired season, and trim to the desired time range.
+    xSea = xSeasons.sel(time=xSeasons.time.dt.month.isin(season_sel))
+    xSea = xSea.sel(time=slice(startDate, endDate))
+    return xSea
+
+###############################################################################
+# Compute desired global seasonal mean using month_to_season()
+
+# Choose the winter season (December-January-February)
+season = "DJF"
+SLP = month_to_season(ds, season)
+print_debug('\n\nSLP:\n\n')
+print_debug(SLP)
+
+# Diagnostic plot: show slice of SLP
+sliceSLP = SLP.sel(lat=slice(latS, latN), lon=slice(lonL, lonR))
+
+print_debug('\n\nsliceSLP:\n')
+print_debug(sliceSLP)
+
+###############################################################################
+# Create weights: sqrt(cos(lat))   [or sqrt(gw) ]
+
+deg2rad = 4. * atan(1.) / 180.
+clat = SLP['lat'].astype(dtype=np.float64)
+clat = np.sqrt(np.cos(deg2rad * clat))
+print_debug('\n\nclat:\n')
+print_debug(clat)
+
+###############################################################################
+# Multiply SLP by weights:
+
+# Xarray uses the supplied coordinate information to apply latitude-based
+# weights to all longitudes and timesteps automatically.
+
+wSLP = SLP
+wSLP['slp'] = clat * SLP['slp']
+
+# For now, metadata for slp must be copied over explicitly; it is not preserved by binary operators like multiplication.
+wSLP['slp'].attrs = ds['slp'].attrs
+wSLP['slp'].attrs['long_name'] = 'Wgt: ' + wSLP['slp'].attrs['long_name']
+
+###############################################################################
+# Subset data to the North Atlantic region:
+
+xw = wSLP.sel(lat=slice(latS, latN), lon=slice(lonL, lonR))
+
+print_debug('\n\nxw:\n\n')
+print_debug(xw.slp)
+
+###############################################################################
+# Compute the EOFs:
+
+eof = eofunc(xw["slp"], neof, time_dim=1, meta=True)
+
+print_debug('\n\neof:\n\n')
+print_debug(eof)
+
+eof_ts = eofunc_ts(xw["slp"], eof, time_dim=1, meta=True)
+
+print_debug('\n\neof_ts:\n\n')
+print_debug(eof_ts)
+
+###############################################################################
+# Normalize time series:
+
+# Sum spatial weights over the area used.
+nLon = len(xw['lon'])
```

Comment:
```suggestion
nLon = xw.sizes["lon"]
```

Reply:
I question whether this is simpler, given that the user needs to know about "sizes".   It seems likely that the user already knows how to calculate the length of an array.

Reply:
I guess the argument for using "sizes" is that length was calculated already, so the re-calculation is wasteful.   Probably worth doing for big arrays.

Reply:
or the user could learn about `sizes` through this script...

Reply:
I see your point; done.